lighthouse-core/test/lib/sentry-test.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+jest.mock('raven');
+
+const raven = require('raven');
+const Sentry = require('../../lib/sentry');
+
+/* eslint-env jest */
+
+describe('Sentry', () => {
+  let configPayload;
+  let originalSentry;
+  let originalMathRandom;
+  let ravenConfigFn;
+  let ravenMergeContextFn;
+  let ravenCaptureExceptionFn;
+  let mathRandomFn;
+
+  beforeEach(() => {
+    configPayload = {
+      url: 'http://example.com',
+      flags: {enableErrorReporting: true},
+      environmentData: {},
+    };
+
+    originalSentry = {...Sentry};
+    originalMathRandom = Math.random;
+    ravenConfigFn = jest.fn().mockReturnValue({install: jest.fn()});
```

Comment:
can this just set these directly (`raven.config = jest.fn().mockReturnValue({install: jest.fn()});`) and then do the `expect()` checks below on the actual thing (`expect(raven.config).toSomethingSomething`). For whatever reason that's easier for me to parse than checking `expect(ravenConfigFn)`

Reply:
or from the docs it seems like you can also do `raven.config.mockReturnValue({install: jest.fn()});` rather than the assignment. Does that work?

Reply:
> can this just set these directly

sure, no prob

> it seems like you can also do raven.config.mockReturnValue({install: jest.fn()}); rather than the assignment. Does that work?

mostly :) the auto-mock only happens when the module is required in for the test, it's not reset for every `it`. Rather than rely on the magic mocks, have reset methods, and do the configuration, it seemed clearer to do the explicit `jest.fn` for the ones we're actually making assertions about. That make sense?